compare_experiments/compare_generic_genes.py

```diff
@@ -0,0 +1,324 @@
+# -
+
+print(gene_ranking_same_combined.shape)
+gene_ranking_same_combined.head()
+
+# Check for NAs
+gene_ranking_same_combined[pd.isnull(gene_ranking_same_combined).any(axis=1)]
+
+# +
+# Plot correlation between ranking
+fig = sns.jointplot(
+    data=gene_ranking_same_combined,
+    x="Rank 1",
+    y="Rank 2",
+    kind="hex",
+    marginal_kws={"color": "white"},
+)
+
+fig.set_axis_labels(
+    f"Ranking in {project_id1}",
+    f"Ranking in {project_id1} different runs",
+    fontsize=14,
+    fontname="Verdana",
+)
+
+output_figure_filename = "concordance_between_same_recount2_templates.svg"
+fig.savefig(
+    output_figure_filename,
+    format="svg",
+    bbox_inches="tight",
+    transparent=True,
+    pad_inches=0,
+    dpi=300,
+)
+# -
+
+# **Takeaway:**
+# * Running SOPHIE twice using the same template experiment will generate 2 different sets of simulated experiments.
+# * Since the template experiment is the same, these 2 sets of simulated experiments will have the same experimental design structure/biological context
+# * As expected, the concordance is very high especially for high ranked and low ranked genes. The genes in the middle rank are more sensitive to changes so you don't get as clear of a signal compared to the extreme ranked genes.
+
+# ## Correlation between rankings between 2 different experiments
+#
+# Here we compare gene ranking generated by SOPHIE using 2 different template experiments.
+
+# Load gene ranking
+gene_ranking_summary2 = pd.read_csv(
+    gene_ranking_filename2, sep="\t", index_col=0, header=0
+)
+
+# +
+# Get simulated ranking
+gene_ranking1 = gene_ranking_summary1["Rank (simulated)"].rename("Rank 1")
+gene_ranking2 = gene_ranking_summary2["Rank (simulated)"].rename("Rank 2")
+
+# Combine ranking
+gene_ranking_diff_combined = pd.concat([gene_ranking1, gene_ranking2], axis=1)
+# -
+
+print(gene_ranking_diff_combined.shape)
+gene_ranking_diff_combined.head()
+
+# Check for NAs
+gene_ranking_diff_combined[pd.isnull(gene_ranking_diff_combined).any(axis=1)]
+
+# +
+# Plot correlation between ranking
+fig = sns.jointplot(
+    data=gene_ranking_diff_combined,
+    x="Rank 1",
+    y="Rank 2",
+    kind="hex",
+    marginal_kws={"color": "white"},
+)
+
+fig.set_axis_labels(
+    f"Ranking in {project_id1}",
+    f"Ranking in {project_id2}",
+    fontsize=14,
+    fontname="Verdana",
+)
+
+output_figure_filename = "concordance_between_diff_recount2_templates.svg"
+fig.savefig(
+    output_figure_filename,
+    format="svg",
+    bbox_inches="tight",
+    transparent=True,
+    pad_inches=0,
+    dpi=300,
+)
+# -
+
+# **Takeaway:**
+#
+# * Looks like there is good concordance between highly ranked genes (i.e. generic genes)
+# * By comparison if we run SOPHIE using two different template experiments, there are genes in the off-diagonal regions that might indicate that there are generic within the given context of the specific experiment.
```

Comment:
What is the difference between a gene that is generic within the context of a specific experiment and a gene that is just upregulated within that experiment? Is it generic in that it appears across several seeds?

Reply:
Generic genes are those that are DE, differentially up or down regulated across multiple experiments/seeds. So the idea is that some gene experiment designs or biological contexts, could in theory yield different generic genes (i.e. genes that are frequently DE in a cancer context are not necessarily the same genes that are frequently DE in the context of some other stimuli)